typegraph/core/src/utils/fs_host.rs

```diff
@@ -214,3 +216,64 @@ pub fn hash_file(path: &Path) -> Result<(String, u32), String> {
 pub fn path_exists(path: &Path) -> Result<bool, String> {
     path_exists_host(&path.to_string_lossy())
 }
+
+pub fn is_glob(path: &str) -> bool {
+    let path = PathBuf::from(path);
+    if let Some(base_name) = path.file_name() {
+        let base_name = base_name.to_str().unwrap();
+        if base_name.contains('*') || base_name.contains('?') {
+            return true;
+        }
+    }
+
+    false
+}
+
+pub fn get_dir(path: &str) -> PathBuf {
+    let directory = PathBuf::from(path);
+
+    PathBuf::from(directory.parent().unwrap())
+}
+
+pub fn get_matching_files(glob_pattern: &str) -> Result<Vec<PathBuf>, String> {
+    let mut matching_files = vec![];
+    let glob_dir = get_dir(glob_pattern);
+
+    let glob_abs_path = make_absolute(&glob_dir)?;
+    let all_files = expand_path(&glob_abs_path, &[])?;
+    let glob_pattern = GlobPattern::new(glob_pattern).unwrap();
+    // eprint(&format!("++++++++++++++++++ {}", glob_pattern.as_str()));
+    for file in all_files {
+        if glob_pattern.matches(file.to_str().unwrap()) {
+            matching_files.push(make_relative(&file)?);
+        }
+    }
+
+    // eprint(&format!("***************** {:?}", matching_files));
+    Ok(matching_files)
+}
+
+pub fn resolve_globs_dirs(deps: Vec<String>) -> Result<Vec<PathBuf>, String> {
+    let mut resolved_deps = vec![];
```

Comment:
Maybe use a HashSet? Each globs inside `deps` may expand to similar paths.

Reply:
good point! they get filtered later on the pipeline but can also do it here.